src/client.rs

```diff
@@ -256,24 +253,54 @@ impl<'a> RequestBuilder<'a> {
 
 /// A Response to a submitted `Request`.
 #[derive(Debug)]
-pub struct Response {
-    inner: ::hyper::client::Response,
+pub enum Response {
+    /// A Response to a submitted `Request`.
+    PlainText(::hyper::client::Response),
+    /// A Response to a submitted `Request`.
+    Gzip(::hyper::client::Response),
 }
 
 impl Response {
+    /// Constructs a response from a hyper request
+    /// Uses the correct variant by inspecting the content-encoding header.
+    pub fn from_hyper_response(res: ::hyper::client::Response) -> Self {
+        let mut is_gzip = false;
+        match res.headers.get::<ContentEncoding>() {
```

Comment:
As mentioned before, we should be checking `TransferEncoding`, not `ContentEncoding`.